File: z-clients/javascript/src/models/msgFifoReceiveOut.ts
// this file is @generated
import {
    type FifoMsgOut,
    FifoMsgOutSerializer,
} from './fifoMsgOut';

export interface MsgFifoReceiveOut {
    msgs: FifoMsgOut[];
}

export const MsgFifoReceiveOutSerializer = {
    // biome-ignore lint/suspicious/noExplicitAny: intentional any
    _fromJsonObject(object: any): MsgFifoReceiveOut {
        return {
            msgs: object['msgs'].map((item: FifoMsgOut) => FifoMsgOutSerializer._fromJsonObject(item)),
        };
    },

    // biome-ignore lint/suspicious/noExplicitAny: intentional any
    _toJsonObject(self: MsgFifoReceiveOut): any {
        return {
            'msgs': self.msgs.map((item) => FifoMsgOutSerializer._toJsonObject(item)),
        };
    }
}

File: z-clients/rust/src/models/msg_fifo_receive_out.rs
// this file is @generated
use serde::{Deserialize, Serialize};

use super::fifo_msg_out::FifoMsgOut;

#[derive(Clone, Debug, Deserialize, Serialize)]
pub struct MsgFifoReceiveOut {
    pub msgs: Vec<FifoMsgOut>,
}

impl MsgFifoReceiveOut {
    pub fn new(msgs: Vec<FifoMsgOut>) -> Self {
        Self { msgs }
    }
}
